services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,17 +1,49 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
-import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
+import cromwell.pipeline.datastorage.dto.{
+  FileParameter,
+  ProjectConfiguration,
+  ProjectFileConfiguration,
+  ProjectId,
+  TypedValue
+}
 
+import java.nio.file.Path
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
-    repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
-      .map(_.toString)
+  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] = {
+    def updateConfiguration(configuration: ProjectConfiguration) =
+      repository.updateOne(toDocument(configuration), "projectId", projectConfiguration.projectId.value).map(_.toString)
+
+    def toMap(config: ProjectFileConfiguration): Map[Path, Map[String, TypedValue]] =
```

Comment:
I thought you will use a Map as a main data structure and get rid of this method but if you want to keep it then let's make it more readable.
Can't we just return `(Path, Map[String, TypedValue])` ? Usually we should not return tuples because they make code unreadable but in this case `Map[Path, Map[String, TypedValue]]` is no better than `(Path, Map[String, TypedValue])`